### PR TITLE
fix: Change RBAC enums

### PR DESCRIPTION
**Pull request by fellipeh**
# Overview

This PR is being created to address [ESSNTL-4505](https://issues.redhat.com/browse/ESSNTL-4505).

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -50,10 +50,14 @@
 
 
 class Permission(Enum):
-    READ = "inventory:hosts:read"
-    WRITE = "inventory:hosts:write"
-    ADMIN = "inventory:*:*"
+    HOSTS_READ = "inventory:hosts:read"
+    HOSTS_WRITE = "inventory:hosts:write"
+    HOSTS_ADMIN = "inventory:hosts:*:*"
```
Comment (fstavela) [inline app/__init__.py:55]: Is this a valid permissions? I've never seen a permission like this. And what would be the difference between this and `HOSTS_ALL`?

Comment (fellipeh) [reply app/__init__.py:55]: Thinking here make no sense to have `HOSTS_ADMIN` and `GROUPS_ADMIN` since we already have ALL for both cases.. makes more sense to have only ADMIN.. I'll change here.

**Review by thearifismail — COMMENTED** [review]
Since I never worked on `rbac`, I never noticed the existence of `rbac-mock-data` in `utility` folder.  Shouldn't `rback-mock-data` be located in `tests/helpers` folder?

File: lib/middleware.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def modified_func(*args, **kwargs):
 
             for rbac_permission in rbac_data:
                 if (
-                    rbac_permission["permission"] == Permission.ADMIN.value  # inventory:*:*
+                    rbac_permission["permission"] == Permission.HOSTS_ADMIN.value  # inventory:*:*
```
[the remark this answers is not in the document]
Comment (fellipeh) [reply lib/middleware.py]: This was changed on my last commit 😃

File: api/host_group.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
 
 @api_operation
-@rbac(Permission.WRITE)
+@rbac(Permission.HOSTS_WRITE)
```
Comment (fstavela) [inline api/host_group.py:25]: I think `GROUPS_WRITE` is more appropriate here, since this is a Groups endpoint.

File: api/unleash.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 
 @api_operation
-@rbac(Permission.READ)
+@rbac(Permission.HOSTS_READ)
```
Comment (fstavela) [inline api/unleash.py:11]: This probably doesn't matter, but `GROUPS_READ` would probably be better here.

Maybe not even that, maybe there are some RBAC roles for feature flags that we should use here instead. I don't know.

Comment (kruai) [reply api/unleash.py:11]: I actually have no idea what kind of permission would make the most sense here. Should it be a secret to know whether we have a FF enabled or not?

**Comment by thearifismail** [comment]
> Since I never worked on `rbac`, I never noticed the existence of `rbac-mock-data` in `utility` folder. Shouldn't `rback-mock-data` be located in `tests/helpers` folder?

To check it, I suggest moving the folder to `tests/helpers` folder and run tests.  You may have to update the path in test files.  The only reason I can think of putting rback-mocks in the utility folder could issues with python paths resolution, which I hope is not the case.

File: api/unleash.py (hunk 4 above)
Comment (fstavela) [reply api/unleash.py]: That's a good point. So maybe we don't need any RBAC restrictions on that endpoint?

**Comment by kruai** [comment]
This had all PR checks pass, so I'm going to merge it once it's done rebasing.

File: api/unleash.py (hunk 4 above)
Comment (kruai) [reply api/unleash.py]: I think removing the restriction makes the most sense, yeah.

Comment (kruai) [reply api/unleash.py]: @fellipeh This conversation was marked as resolved, but the issue was not addressed

**Review by thearifismail — DISMISSED** [review]
LGTM

**Review by kruai — APPROVED** [review]

**Comment by kruai** [comment]
The PR checks passed earlier, so I'm going to go ahead and merge. The PR check queue is really long rn
